pkg/klusterlet/clusterinfo/loggingInfo_sync.go

```diff
@@ -2,329 +2,60 @@ package controllers
-		return nil, fmt.Errorf("agent service should be in type of loadbalancer")
-	}
-
-	if len(klSvc.Status.LoadBalancer.Ingress) == 0 {
-		return nil, fmt.Errorf("agent load balancer service does not have valid ip address")
-	}
-
-	// Only get the first IP and host name
-	endpoint := &corev1.EndpointAddress{
-		IP:       klSvc.Status.LoadBalancer.Ingress[0].IP,
-		Hostname: klSvc.Status.LoadBalancer.Ingress[0].Hostname,
-	}
-
-	return endpoint, nil
-}
-
-func (s *loggingInfoSyncer) getEndpointAddressFromRoute(ctx context.Context, isOCP311 bool) (*corev1.EndpointAddress, error) {
-	var (
-		err      error
-		endpoint = &corev1.EndpointAddress{}
-	)
-
-	// when we delete addon agent deployment, we expect the route is deleted too.
-	owner, err := s.getOwnerRef(ctx, isOCP311)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get owner. error: %v", err)
-	}
-	newRoute := s.newRoute(*owner)
-	// there is no configIngress in OCP 311,only get domain from configIngress in OCP4.
-	// for OCP311 the host is empty and will be generated by OCP.
-	if !isOCP311 {
-		ingressDomain, err := s.getIngressDomain(ctx)
+	errs := []error{}
+	_, err := s.managementClusterClient.CoreV1().Services(s.agentNamespace).Get(ctx, s.agentName, metav1.GetOptions{})
+	if err == nil {
+		err = s.managementClusterClient.CoreV1().Services(s.agentNamespace).Delete(ctx, s.agentName, metav1.DeleteOptions{})
 		if err != nil {
-			return nil, fmt.Errorf("failed to get ingress domain.error %v", err)
+			errs = append(errs, err)
 		}
-		hostName := subHostName(s.agentNamespace)
-		newRoute.Spec.Host = fmt.Sprintf("%s.%s", hostName, ingressDomain)
+	} else if !errors.IsNotFound(err) {
+		errs = append(errs, err)
 	}
 
-	route, err := s.routeV1Client.RouteV1().Routes(s.agentNamespace).Get(ctx, s.agentName, metav1.GetOptions{})
-	switch {
-	case errors.IsNotFound(err):
-		route, err = s.routeV1Client.RouteV1().Routes(s.agentNamespace).Create(ctx, newRoute, metav1.CreateOptions{})
-		if err != nil {
-			return nil, fmt.Errorf("failed to create route.error %v", err)
-		}
-	case err != nil:
-		return nil, fmt.Errorf("failed to get route. error %v", err)
-	case needUpdateRoute(route, newRoute):
-		route, err = s.routeV1Client.RouteV1().Routes(s.agentNamespace).Update(ctx, route, metav1.UpdateOptions{})
-		if err != nil {
-			return nil, fmt.Errorf("failed to update route. error %v", err)
-		}
+	if clusterInfo.Status.DistributionInfo.Type != clusterv1beta1.DistributionTypeOCP {
+		return utils.NewMultiLineAggregate(errs)
 	}
 
-	endpoint.Hostname = route.Spec.Host
-	return endpoint, nil
-}
-
-func (s *loggingInfoSyncer) getOwnerRef(ctx context.Context, ocp311 bool) (*metav1.OwnerReference, error) {
-	if ocp311 {
-		deploy, err := s.managementClusterClient.ExtensionsV1beta1().Deployments(s.agentNamespace).Get(ctx, s.agentName, metav1.GetOptions{})
+	_, err = s.routeV1Client.RouteV1().Routes(s.agentNamespace).Get(ctx, s.agentName, metav1.GetOptions{})
```

Comment:
I am not sure if it is needed. I do not think we need to delete the route. This could be doced and let user do it manually.